src/main/java/org/opensearch/security/filter/SecurityRestFilter.java

```diff
@@ -138,19 +148,56 @@ private boolean userIsSuperAdmin(User user, AdminDNs adminDNs) {
         return user != null && adminDNs.isAdmin(user);
     }
 
+    private boolean authorizeRequest(RestHandler original, RestRequest request, RestChannel channel, User user) throws Exception {
+        if (original instanceof RestSendToExtensionAction) {
+            List<RestHandler.Route> extensionRoutes = original.routes();
+            Optional<RestHandler.Route> handler = extensionRoutes.stream()
+                    .filter(rh -> rh.getMethod().equals(request.method()))
+                    .filter(rh -> restPathMatches(request.path(), rh.getPath()))
+                    .findFirst();
+            if (handler.isPresent() && handler.get() instanceof ProtectedRoute) {
+                String action = ((ProtectedRoute)handler.get()).name();
+                PrivilegesEvaluatorResponse pres = evaluator.evaluate(user, action);
```

Comment:
I know this is a  draft so maybe you have not gotten there yet, but I thought that the evaluator only operates based on role resolution. If this is the case, what is your plan for hosting the REST routes in this mechanism? Are you making each extension add a role with the route as a permission string or are we going to add a new role used to hold these routes or something else?